src/packagedcode/podspec.py

```diff
@@ -0,0 +1,271 @@
+# The ScanCode software is licensed under the Apache License version 2.0.
+# Data generated with ScanCode require an acknowledgment.
+# ScanCode is a trademark of nexB Inc.
+#
+# You may not use this software except in compliance with the License.
+# You may obtain a copy of the License at: http://apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software distributed
+# under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+# CONDITIONS OF ANY KIND, either express or implied. See the License for the
+# specific language governing permissions and limitations under the License.
+#
+# When you publish or redistribute any data created with ScanCode or any ScanCode
+# derivative work, you must accompany this data with the following acknowledgment:
+#
+#  Generated with ScanCode and provided on an "AS IS" BASIS, WITHOUT WARRANTIES
+#  OR CONDITIONS OF ANY KIND, either express or implied. No content created from
+#  ScanCode should be considered or used as legal advice. Consult an Attorney
+#  for any legal advice.
+#  ScanCode is a free software code scanning tool from nexB Inc. and others.
+#  Visit https://github.com/nexB/scancode-toolkit/ for support and download.
+
+from __future__ import absolute_import
+from __future__ import print_function
+from __future__ import unicode_literals
+
+from collections import OrderedDict
+import csv
+import glob
+import io
+import logging
+import os
+import re
+
+import attr
+from packageurl import PackageURL
+
+from commoncode.fileutils import py2
+from commoncode import filetype
+from commoncode import fileutils
+from packagedcode import models
+
+
+"""
+Handle Podspec Package Manager
+"""
+
+TRACE = False
+
+logger = logging.getLogger(__name__)
+
+if TRACE:
+    import sys
+    logging.basicConfig(stream=sys.stdout)
+    logger.setLevel(logging.DEBUG)
+
+
+@attr.s()
+class PodspecPackage(models.Package):
+    metafiles = ('*.podspec',)
+    extensions = ('.podspec',)
+    default_type = 'podspec'
+    default_primary_language = 'Swift'
+    default_web_baseurl = None
+    default_download_baseurl = None
+    default_api_baseurl = None
+
+    @classmethod
+    def recognize(cls, location):
+        yield parse(location)
+
+
+def is_podspec(location):
+    """
+    Checks is the file is a podspec file or not.
+    """
+    return (filetype.is_file(location) and location.endswith('.podspec'))
+
+
+def parse(location):
+    """
+    Return a Package object from a .podspec file or None.
+    """
+    if not is_podspec(location):
+        return
+
+    package_data = parse_podspec(location)
+    return build_package(package_data)
+
+
+def build_package(podspec_data):
+    """
+    Return a Pacakge object from a package data mapping or None.
+    """
+    name = podspec_data.get('name')
+    version = podspec_data.get('version')
+    description = podspec_data.get('description')
+    homepage_url = podspec_data.get('homepage')
```

Comment:
What happens when these are not defined? We should have a test for this also

Reply:
Then it will send None. I think script will also be working fine when these are not defined.  I will add test for this also. :)

Reply:
Added test for this too :)